apis/database/v1alpha1/rds_types.go

```diff
@@ -0,0 +1,169 @@
+// +kubebuilder:resource:scope=Cluster
+type RDSInstanceClass struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// SpecTemplate is a template for the spec of a dynamically provisioned
+	// RDSInstance.
+	SpecTemplate RDSInstanceClassSpecTemplate `json:"specTemplate"`
+}
+
+// +kubebuilder:object:root=true
+
+// RDSInstanceList contains a list of RDSInstance
+type RDSInstanceList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []RDSInstance `json:"items"`
+}
+
+// +kubebuilder:object:root=true
+
+// An RDSInstance is a managed resource that represents an RDS instance.
+// +kubebuilder:printcolumn:name="STATUS",type="string",JSONPath=".status.bindingPhase"
+// +kubebuilder:printcolumn:name="STATE",type="string",JSONPath=".status.atProvider.dbInstanceStatus"
+// +kubebuilder:printcolumn:name="CLASS",type="string",JSONPath=".spec.classRef.name"
+// +kubebuilder:printcolumn:name="ENGINE",type="string",JSONPath=".spec.forProvider.engine"
+// +kubebuilder:printcolumn:name="VERSION",type="string",JSONPath=".spec.forProvider.engineVersion"
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+type RDSInstance struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   RDSInstanceSpec   `json:"spec"`
+	Status RDSInstanceStatus `json:"status,omitempty"`
+}
+
+// An RDSInstanceClassSpecTemplate is a template for the spec of a dynamically
+// provisioned RDSInstance.
+type RDSInstanceClassSpecTemplate struct {
+	runtimev1alpha1.ClassSpecTemplate `json:",inline"`
+	ForProvider                       RDSInstanceParameters `json:"forProvider"`
+}
+
+// An RDSInstanceSpec defines the desired state of an RDSInstance.
+type RDSInstanceSpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+	ForProvider                  RDSInstanceParameters `json:"forProvider"`
+}
+
+// An RDSInstanceStatus represents the observed state of an RDSInstance.
+type RDSInstanceStatus struct {
+	runtimev1alpha1.ResourceStatus `json:",inline"`
+	AtProvider                     *RDSInstanceObservation `json:"atProvider,omitempty"`
+}
+
+// RDSInstanceParameters define the desired state of an RDS instance.
+type RDSInstanceParameters struct {
+	// Engine is the name of the database engine to be used for this instance.
+	// Engine is a required field.
+	// +immutable
+	Engine string `json:"engine"`
+
+	// EngineVersion indicates the database engine version.
+	EngineVersion string `json:"engineVersion,omitempty"`
+
+	// DBInstanceClass is the machine class of the instance, e.g. "rds.pg.s1.small"
+	DBInstanceClass string `json:"dbInstanceClass,omitempty"`
+
+	// DBInstanceStorageInGB indicates the size of the storage in GB.
+	DBInstanceStorageInGB int `json:"dbInstanceStorageInGB,omitempty"`
```

Comment:
Are these fields required? If so they should not have `omitempty`, if not they should be pointer types per [Kubernetes API conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required).

Reply:
Yes. Required.